### DISCUSSION TITLE
Can I import 10K+ entities from our own DB to Dynamic 365?

### DISCUSSION THREAD

**Discussion by Notaduck** [AleksandrRogov/DynamicsWebApi#154] [Q&A] [answered]
I have had a consultant creating an import service at (scribe) where it would import a lot of entities from our own database, transform and insert them into our dynamics 365 app.

I was wondering if this package is suitable for this? It will probably be a lot cheaper and easier to maintain, but it comes up to if this is ready for production?

**Comment by AleksandrRogov** [comment, marked as answer]
v2 is not ready for production just yet, but v1 is. 
DynamicsWebApi can be used in any Node.js application that integrates with Microsoft Dataverse. It will certainly facilitate the development of said application. How many records that application imports depends on how it was developed.

Just to clarify: DynamicsWebApi is incomparable to Scribe. Scribe has GUI to create integration pipelines that does not require any programming knowledge whereas DynamicsWebApi requires knowledge of JavaScript/Typescript and basic knowledge of Rest API and Dataverse Web API. Thus, if you are planning on hiring developers then you will be fine. Consultants (most probably) won't be able to utilize it.

Let me know if you have more questions.

**Reply by Notaduck** [reply]
That was the exact anwser I was looking for :) Thanks! 

Can I ask when you will think v2 is ready for production?

**Reply by AleksandrRogov** [reply]
@Notaduck great! I was wondering if my answer was good for you or not. glad that it was :)

I am testing v2 right now and the results are great. There weren't any issues with the library itself (and other people that help me with testing said the same). The only problem was with the bundles for the browsers (not node.js applications), which I have already resolved but haven't published the fixed scripts yet. So, I would say, expect v2 to be released very soon, in a couple of weeks most probably.

**Reply by Notaduck** [reply]
That is great to hear! I will only be using it with node, so I will be looking forward to the final release of v2 since it doesn't make sense to start building a project with v1. 

Have a great summer!
